api/v1alpha1/addon_types.go

```diff
@@ -134,40 +143,156 @@ func (ClusterAutoscaler) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
-type CSIProviders struct {
+type DefaultStorage struct {
+	ProviderName           string `json:"providerName"`
+	StorageClassConfigName string `json:"storageClassConfigName"`
+}
+
+type CSI struct {
 	// +optional
 	Providers []CSIProvider `json:"providers,omitempty"`
 	// +optional
-	DefaultClassName string `json:"defaultClassName,omitempty"`
+	DefaultStorage *DefaultStorage `json:"defaultStorage,omitempty"`
 }
 
 type CSIProvider struct {
-	Name string `json:"name,omitempty"`
+	Name string `json:"name"`
+
+	// +optional
+	StorageClassConfig []StorageClassConfig `json:"storageClassConfig,omitempty"`
+
+	Strategy AddonStrategy `json:"strategy"`
+
+	// +optional
+	Credentials *corev1.SecretReference `json:"credentials,omitempty"`
```

Comment:
Right now the responsibility of using the correct format of the secret and then CAREN would need to copy this over to the newly created workload cluster.  We can deliver this to the target workload cluster via ClusterResourceSet.